stable/ambassador/templates/crd-delete.yaml

```diff
@@ -0,0 +1,40 @@
+{{- if and .Values.crds.enabled (not .Values.crds.keep)}}
+apiVersion: batch/v1
+kind: Job
+metadata:
+  name: {{ include "ambassador.fullname" . }}-crd-cleanup
+  namespace: {{ .Release.Namespace }}
+  annotations:
+    "helm.sh/hook": pre-delete
+    "helm.sh/hook-weight": "3"
+    "helm.sh/hook-delete-policy": hook-succeeded
+  labels:
+    app.kubernetes.io/name: {{ include "ambassador.name" . }}
+    app.kubernetes.io/part-of: {{ .Release.Name }}
+    helm.sh/chart: {{ include "ambassador.chart" . }}
+    app.kubernetes.io/instance: {{ .Release.Name }}
+    app.kubernetes.io/managed-by: {{ .Release.Service }}
+spec:
+  template:
+    metadata:
+      name: {{ include "ambassador.fullname" . }}-crd-cleanup
+      labels:
+        app.kubernetes.io/name: {{ include "ambassador.name" . }}
+        app.kubernetes.io/part-of: {{ .Release.Name }}
+        helm.sh/chart: {{ include "ambassador.chart" . }}
+        app.kubernetes.io/instance: {{ .Release.Name }}
+        app.kubernetes.io/managed-by: {{ .Release.Service }}
+    spec:
+      {{- if .Values.rbac.create }}
+      serviceAccountName: {{ include "ambassador.serviceAccountName" . }}
+      {{- end }}
+      containers:
+        - name: kubectl
+          image: "k8s.gcr.io/hyperkube:v1.12.1"
+          command:
+          - /bin/sh
+          - -c
+          - >
+              kubectl delete crds -l service=ambassador
```

Comment:
Is it not better selecting on one of the existing labels (`app.kubernetes.io/name`, etc) than adding another label everywhere? :)

Reply:
I wanted something that is static across all releases to watch for. This will allow any release to remove the CRDs, not just the one that created the CRDs. 

Thinking about it, this may be an issue if someone accidentally sets `crds.keep: false` in a second release and then deletes that release. They will unknowingly delete all CRD resources which will remove all Ambassador configuration.

Do you have an opinion on this? Was the past behavior that only the release that created the CRDs could remove them?

Reply:
Hmm, not sure about how it works right now with multiple releases tbh. But that any release would try to remove CRDs when purged sounds scary 😅 so I'd probably opt for only trying to remove if they match the installed release.

Id probably change the condition for the delete:

```
{{- if and .Values.crds.create (not .Values.crds.keep)}}
```
(`enabled` -> `create`) to avoid accidents :)

Reply:
Yes good catch. Let me check on how it worked before I broke it and replicate that.

Reply:
@Flydiverny So I finally checked and it looks like any release that set `crds.keep: false` could delete the crds. 

I agree this seems like the wrong methodology so I am going to change it the behavior of only the master release can remove the crds.